builder/dockerfile/builder.go

```diff
@@ -196,6 +199,11 @@ func (b *Builder) dispatchDockerfileWithCancellation(dockerfile *parser.Result)
 		// TODO: get this from dispatch
 		imageID = b.image
 
+		if b.Aux != nil {
+			if err := b.Aux.Emit(types.BuildResult{ID: imageID}); err != nil {
+				return "", err
+			}
+		}
```

Comment:
This is going to `Emit()` an ID for every instruction in the Dockerfile, is that what you want?

I don't know if this should be part of dispatch, it might be more appropriate in `api/server/backend/build/backend.go`.